libsql-server/src/replication/snapshot.rs

```diff
@@ -201,17 +276,34 @@ impl SnapshotMerger {
         log_id: Uuid,
     ) -> anyhow::Result<()> {
         let mut snapshots = Self::init_snapshot_info_list(db_path).await?;
-        while let Some((name, size, db_page_count)) = receiver.recv().await {
-            snapshots.push((name, size));
-            if Self::should_compact(&snapshots, db_page_count) {
-                let compacted_snapshot_info =
-                    Self::merge_snapshots(&snapshots, db_path, log_id).await?;
-                snapshots.clear();
-                snapshots.push(compacted_snapshot_info);
+        let mut working = false;
+        let mut job: Pin<Box<dyn Future<Output = anyhow::Result<_>> + Sync + Send>> =
+            Box::pin(std::future::pending());
+        loop {
+            tokio::select! {
+                Some((name, size, db_page_count)) = receiver.recv() => {
+                    snapshots.push((name, size));
+                    if !working && Self::should_compact(&snapshots, db_page_count) {
+                        let snapshots = std::mem::take(&mut snapshots);
+                        let fut = async move {
+                            let compacted_snapshot_info =
+                                Self::merge_snapshots(snapshots, db_path, log_id).await?;
+                            Ok(compacted_snapshot_info)
```

Comment:
I think this should be spawned in the case that we somehow error out of this function/cancel out. If we spawn and cancel out the merge will continue to ensure consistency so I think its just safer to always spawn this job and just track a join handle here?